src/features/vote/helpers/formatVoteDataToCommit.ts

```diff
@@ -22,12 +30,27 @@ export async function formatVoteDataToCommit(
   publicKey: string
 ) {
   const postValues = [] as PostCommitVote[];
+  let newCommits = {} as BackupCommit;
+  const backupCommits = localStorage.getItem("backupCommits");
+  if (backupCommits) {
+    newCommits = { ...JSON.parse(backupCommits) };
+    // To prevent null errors in nested object, need to create these objects on the fly
+    // Based on if the user has voted before or not in a given round or using a given account
+    if (!newCommits[address]) {
+      newCommits[address] = {
+        [roundId]: {},
+      };
+    }
+    if (newCommits[address] && !newCommits[address][roundId]) {
+      newCommits[address][roundId] = {};
+    }
```

Comment:
this is really annoying, dealing with nested vals. theres a couple alternatives to consider. you can use lodash .has and .set to check for existence as well as set data deeply nested, will add to bundle size, but these can be installed as their own packages.

another option is to flatten the keys into a unique id and keep the object only 1 level deep.  You'd probably want a helper to generate the id based on the props you need to nest.  this is also nice since it plays well with localstorage.

just wanted to propose some things since it seems like most of the lines are dedicated to managing nested object

Reply:
Yep, nested objects suck. I'll look into if we can make this better.